### ISSUE TITLE
Would it work for SD2?

### ISSUE THREAD

**Issue by KovenYu** [InstantStyle/InstantStyle#15]
Thanks for the great work. would there be SD2.0 version?

**Comment by ResearcherXman** [comment]
But SD2.0 is not a common adopted model, why do you need it?

**Comment by KovenYu** [comment]
Thank you for your prompt reply! I'm using SD2.0-inpainting in my research projects (such as https://kovenyu.com/wonderjourney/), as I found that among the inpainting models (SD1.5, SD2.0, SDXL), SD2.0 works way better than the other two. I wonder if there can be a plug-and-play style control for it.

Thanks!

**Comment by ResearcherXman** [comment]
@KovenYu We haven't tested on inpainting model yet, although they have the same structure except for the first input channel. Are you using [stable-diffusion-2-inpainting](https://huggingface.co/stabilityai/stable-diffusion-2-inpainting)? Or [stabilityai/stable-diffusion-2](https://huggingface.co/stabilityai/stable-diffusion-2)? I believe they both can do inpainting.

**Comment by KovenYu** [comment]
Yeah. I have been using SD2-inpainting. Their architectures are basically the same. I wonder if I can apply your method to SD2-inpainting.

**Comment by ResearcherXman** [comment]
We will check it today.

**Comment by KovenYu** [comment]
Thank you!

**Comment by ResearcherXman** [comment]
For [SDXL-Inpainting](https://huggingface.co/diffusers/stable-diffusion-xl-1.0-inpainting-0.1), it works as below. But as IP-Adapter haven't released checkpoint for SD2.0, we need to retrain it. The risk here is that we find the SD1.5 has less style control, thus we are not sure whether SD2.0 works well. However, if you really need it, we can discuss for research cooperation. @KovenYu 

<img width="1193" alt="截屏2024-04-08 16 17 43" src="https://github.com/InstantStyle/InstantStyle/assets/153269709/aeefd9c5-de1c-4ced-b0eb-7e7a6c356e58">

**Comment by scarbain** [comment]
Just a quick note, we might soon have a great inpainting model from BrushNet authors. Their inpainting model works great on  SD1.5 and they're currently training SDXL.
https://github.com/TencentARC/BrushNet
